src/transaction/TransactionType.ts

```diff
@@ -1,24 +1,27 @@
-import { Transaction } from './Transaction'
-
+import { TokenTransactionV2 } from '../types'
 export interface Context {
   userAddress: string
   tokens?: string[]
 }
 
-export abstract class TransactionType {
+export abstract class TransactionType<T> {
   protected readonly context!: Context
 
   constructor(context: Context) {
     this.context = context
   }
 
-  abstract matches(transaction: Transaction): boolean
-  abstract getEvent(transaction: Transaction): Promise<any>
-  abstract isAggregatable(): boolean
+  abstract matches(transaction: T): boolean
+  abstract getEvent(transaction: T): Promise<TokenTransactionV2>
+
+  // Since this is a no-op for some transaction types, we default it to false.
+  isAggregatable(): boolean {
+    return false
+  }
```

Comment:
I think the default to false is just confusing and not obvious when looking at the implementations.

If you do decide to make it back into an abstract, it looks like you could use an abstract property instead
```suggestion
  abstract readonly isAggregatable: boolean
```

Reply:
Yeah, aggregation is only required for Blockscout transactions, so I didn't want Alchemy transfer types to be forced to define it. I've subclassed this TransactionType for Blockscout and required that to have an abstract property instead.